cerberus/validator.py

```diff
@@ -1038,8 +1058,17 @@ def _validate_keyschema(self, schema, field, value):
     def _validate_readonly(self, readonly, field, value):
         """ {'type': 'boolean'} """
         if readonly:
-            self._error(field, errors.READONLY_FIELD)
-            return True
+            if not self.is_normalized:
+                self._error(field, errors.READONLY_FIELD)
+                return True
+            # If the document was normalized (and therefore already been
+            # checked for readonly fields), we still have to return True
+            # if an error was filed if readonly is a priority rule.
+            if 'readonly' in self.priority_validations:
+                has_error = self.schema_error_tree.fetch_errors_from(
```

Comment:
I did not mean my understanding of both trees, but the code readability at that specific location. But we don't need a deep discussion about this small issue, I am happy to go with your suggestion as well - you are way more experienced with the Cerberus codebase than I am, so I trust you with that.